File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

bool solution(const char* s) {
    if (s[0] == ')' || s[strlen(s)-1] == '(') {
        return false;
    } 
    int check = 0; 
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] == '(') {
            check++;
        } else {
            check--;
        }
        
        if (check < 0) {
            return false;
        }
    }
    if (!check) {
        return true;
    } else {
        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    let steak = [];
    
    for (let i of s) {
        if (i === '(') {
            steak.push(i);
        } else {
            if (steak.length === 0) {
                return false;
            }
            steak.pop()
        }
    }
    
    if (steak.length !== 0) {
        return false;
    }
    
    return true;
}